src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -888,6 +888,14 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                         if (body != null && ((methodSymbol.ContainingType.IsStructType() && !methodSymbol.IsImplicitConstructor) || instrumentForDynamicAnalysis))
                         {
+                            if (instrumentForDynamicAnalysis && methodSymbol.IsImplicitConstructor)
```

Comment:
The condition is more correct as written. Were it to be changed as indicated above, then field initializers would get analyzed twice for explicit constructors of classes (once here, and once in FlowAnalysisPass.Rewrite).

I agree that the logic is hard to follow, but I'm not sure either that now is the best time to rearrange the code or that I am the most appropriate person to do it.